ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -216,6 +246,34 @@ private MessageWriteSetStateInStore checkWriteSetStateInStore(MessageWriteSet me
     }
   }
 
+  /**
+   * Checks the used capacity of the store against the configured percentage thresholds to see if the store
+   * should be read-only or read-write
+   * @param totalCapacity total capacity of the store in bytes
+   * @param usedCapacity total used capacity of the store in bytes
+   */
+  private void checkCapacityAndUpdateHelix(long totalCapacity, long usedCapacity) {
+    if (clusterManagerWriteStatusDelegate != null) {
+
+      int threshold = config.storeReadOnlyEnableSizeThresholdPercentage;
+      int delta = config.storeReadWriteEnableSizeThresholdPercentageDelta;
+      double percentFilled = (index.getLogUsedCapacity() / (double) log.getCapacityInBytes()) * 100;
```

Comment:
Perhaps not a big deal, but you could compute `thresholdBytesHighWaterMark` one time at startup (`threshold * log.getCapacityInBytes()`) and `thresholdBytesLowWaterMark` (`(threshold - delta) * log.getCapacityInBytes()`) - so that these checks that happen for every event is quicker. Again, not a big deal in the overall scheme of things ...

Reply:
changed